includes/admin/extensions/class-extension-manager.php

```diff
@@ -0,0 +1,495 @@
+
+		// Do not allow WordPress to search/download translations, as this will break JS output.
+		remove_action( 'upgrader_process_complete', array( 'Language_Pack_Upgrader', 'async_upgrade' ), 20 );
+
+		// Create the plugin upgrader with our custom skin.
+		$installer = new \EDD\Admin\Installers\PluginSilentUpgrader( new \EDD\Admin\Installers\Install_Skin() );
+
+		// Error check.
+		if ( ! method_exists( $installer, 'install' ) || empty( $plugin ) ) {
+			wp_send_json_error( $result );
+		}
+
+		$installer->install( $plugin ); // phpcs:ignore
+
+		// Flush the cache and return the newly installed plugin basename.
+		wp_cache_flush();
+
+		$plugin_basename = $installer->plugin_info();
+
+		// Check for permissions.
+		if ( ! current_user_can( 'activate_plugins' ) ) {
+			$result['message'] = 'plugin' === $type ? esc_html__( 'Plugin installed.', 'easy-digital-downloads' ) : esc_html__( 'Extension installed.', 'easy-digital-downloads' );
+
+			wp_send_json_error( $result );
+		}
+
+		$this->activate( $plugin_basename );
+	}
+
+	/**
+	 * Activates an existing extension.
+	 *
+	 * @since 2.11.x
+	 * @param string $plugin_basename Optional: the plugin basename.
+	 */
+	public function activate( $plugin_basename = '' ) {
+
+		$result = array(
+			'message'      => __( 'There was an error while performing your request.', 'easy-digital-downloads' ),
+			'is_activated' => false,
+		);
+
+		// Check for permissions.
+		if ( ! check_ajax_referer( 'edd_extensionmanager', 'nonce', false ) || ! current_user_can( 'activate_plugins' ) ) {
+			$result['message'] = __( 'Plugin activation is not available for you on this site.', 'easy-digital-downloads' );
+			wp_send_json_error( $result );
+		}
+
+		$already_installed = false;
+		if ( empty( $plugin_basename ) ) {
+			$plugin_basename   = ! empty( $_POST['plugin'] ) ? sanitize_text_field( $_POST['plugin'] ) : '';
+			$already_installed = true;
+		}
+
+		$plugin_basename = sanitize_text_field( wp_unslash( $plugin_basename ) );
+		$type            = ! empty( $_POST['type'] ) ? sanitize_text_field( $_POST['type'] ) : '';
+		if ( 'plugin' !== $type ) {
+			$type = 'extension';
+		}
+		$result = array(
+			/* translators: "extension" or "plugin" as defined by $type */
+			'message'      => sprintf( __( 'Could not activate the %s.', 'easy-digital-downloads' ), esc_html( $type ) ),
+			'is_activated' => false,
+		);
+		if ( empty( $plugin_basename ) || empty( $type ) ) {
+			wp_send_json_error( $result );
+		}
+
+		$result['basename'] = $plugin_basename;
+
+		// Activate the plugin silently.
+		$activated = activate_plugin( $plugin_basename );
+
+		if ( ! is_wp_error( $activated ) ) {
+
+			/**
+			 * Fire after plugin activating via the EDD installer.
+			 *
+			 * @since 2.11.x
+			 *
+			 * @param string $plugin_basename Path to the plugin file relative to the plugins directory.
+			 */
+			do_action( 'edd_plugin_activated', $plugin_basename );
```

Comment:
Is this hook needed for anything?

Reply:
I had it in back when I wanted to delete the saved options, and it was in the original code I was using as an example/starting point.

Reply:
If we don't have a use case for it right now then I vote we remove it. Can always add it back in later, but once it's in there we can't really take it out. :(